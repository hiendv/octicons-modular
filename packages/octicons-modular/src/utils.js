const assign = (t, ...sources) => {
  for (var s, i = 0, n = sources.length; i < n; i++) {
    s = sources[i]
    for (var p in s) {
      if (!Object.prototype.hasOwnProperty.call(s, p)) {
        continue
      }

      t[p] = s[p]
    }
  }
  return t
}

export { assign }
