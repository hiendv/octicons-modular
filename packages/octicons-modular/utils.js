import chalk from 'chalk'
import path from 'path'
import buble from 'rollup-plugin-buble'
import { uglify } from 'rollup-plugin-uglify'
import postcss from 'rollup-plugin-postcss'
import resolve from 'rollup-plugin-node-resolve'
import postcssImport from 'postcss-import'

import config from './config'

export function info (text) {
  // eslint-disable-next-line no-console
  console.log(chalk.white.bgBlue(text))
}

export function success (text) {
  // eslint-disable-next-line no-console
  console.log(chalk.white.bgGreen(text))
}

export function infoStd (text) {
  process.stdout.write(chalk.white.bgBlue(text))
}

export function successStd (text) {
  process.stdout.write(chalk.white.bgGreen(text))
}

export function rollupIconConfig (icon) {
  return [
    {
      input: icon.path,
      output: [ {
        format: 'cjs',
        file: path.resolve(config.paths.destIcons, icon.file),
        interop: false
      } ],
      external: id => /\/octicon\.js/.test(id)
    },
    {
      input: icon.path,
      output: [ {
        format: 'es',
        file: path.resolve(config.paths.destIconsES, icon.file),
        interop: false
      } ],
      external: id => /\/octicon\.js/.test(id)
    }
  ]
}

export function rollupMainConfig () {
  let input = config.paths.srcMain
  return [
    {
      input: config.paths.srcIconBase,
      output: {
        file: config.paths.destIconBase,
        format: 'cjs',
        exports: 'default'
      },
      plugins: [ buble() ]
    },
    {
      input: config.paths.srcIconBase,
      output: {
        file: config.paths.destIconBaseES,
        format: 'es',
        exports: 'default'
      },
      plugins: [ buble() ]
    },
    {
      input,
      output: {
        file: config.paths.destMainUMD,
        format: 'umd',
        exports: 'named',
        name: config.name
      },
      plugins: [
        resolve(),
        postcss({
          plugins: [ postcssImport() ],
          minimize: true
        }),
        buble(),
        uglify()
      ]
    },
    {
      input,
      output: {
        file: config.paths.destMain,
        format: 'cjs',
        exports: 'named',
        interop: false
      },
      plugins: [
        postcss({
          plugins: [ postcssImport() ],
          minimize: true,
          extract: true
        }),
        buble()
      ],
      external: id => new RegExp(`${config.paths.iconsDir}/`).test(id)
    },
    {
      input,
      output: {
        file: config.paths.destMainES,
        format: 'es'
      },
      plugins: [
        postcss({
          plugins: [ postcssImport() ],
          minimize: true,
          extract: false,
          inject: false
        }),
        buble()
      ],
      external: id => new RegExp(`${config.paths.iconsDir}/`).test(id)
    }
  ]
}

export default { info, infoStd, success, successStd, rollupIconConfig, rollupMainConfig }
