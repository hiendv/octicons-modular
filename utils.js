import chalk from 'chalk'
import cssnano from 'cssnano'
import path from 'path'
import postcss from 'postcss'
import sass from 'rollup-plugin-sass'
import uglify from 'rollup-plugin-uglify'

import config from './config'

export function info (text) {
  // eslint-disable-next-line no-console
  console.log(chalk.white.bgBlue(text))
}

export function success (text) {
  // eslint-disable-next-line no-console
  console.log(chalk.white.bgGreen(text))
}

export function infoStd (text) {
  process.stdout.write(chalk.white.bgBlue(text))
}

export function successStd (text) {
  process.stdout.write(chalk.white.bgGreen(text))
}

export function rollupIconConfig (icon) {
  return {
    input: icon.path,
    output: [ {
      format: 'cjs',
      file: path.resolve(config.paths.destIcons, icon.file)
    }, {
      format: 'es',
      file: path.resolve(config.paths.destIconsES, icon.file)
    } ],
    plugins: [ uglify() ],
    external: id => /\/octicon\.js/.test(id)
  }
}

export function rollupMainConfig () {
  let input = config.paths.srcMain
  let plugins = [
    sass({
      options: { includePaths: [ path.join(__dirname, 'node_modules') ] },
      processor: css => postcss([ cssnano() ]).process(css, { from: void 0 }).then(result => result.css),
      output: config.paths.destMainStyle
    }),
    uglify()
  ]

  return [ {
    input: config.paths.srcIconBase,
    output: {
      file: config.paths.destIconBaseES,
      format: 'es'
    },
    plugins
  }, {
    input: config.paths.srcIconBase,
    output: {
      file: config.paths.destIconBase,
      format: 'cjs',
      exports: 'named'
    },
    plugins
  }, {
    input,
    output: {
      file: config.paths.destMainES,
      format: 'es'
    },
    plugins,
    external: id => new RegExp(`${config.paths.iconsDir}/`).test(id)
  }, {
    input,
    output: {
      file: config.paths.destMain,
      format: 'cjs',
      exports: 'named'
    },
    plugins,
    external: id => new RegExp(`${config.paths.iconsDir}/`).test(id)
  }, {
    input,
    output: {
      file: config.paths.destMainUMD,
      format: 'umd',
      exports: 'named',
      name: config.name
    },
    plugins
  } ]
}

export default { info, infoStd, success, successStd, rollupIconConfig, rollupMainConfig }
