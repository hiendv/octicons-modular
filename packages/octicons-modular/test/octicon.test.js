import factory from '../src/octicon.js'
import { t } from './utils.js'

describe('octicon', () => {
  t(factory('name', 16, 16, {}))
})

describe('octicons', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  test('works with a fake factory', () => {
    jest.doMock(`../lib/octicon.js`, () => {
      return (name, data) => {
        return name
      }
    })
    const Octicons = require('../lib/main.js')
    jest.unmock(`../lib/octicon.js`)

    expect(Octicons.alert).toBe('alert')
    expect(Octicons.zap).toBe('zap')
  })
})
