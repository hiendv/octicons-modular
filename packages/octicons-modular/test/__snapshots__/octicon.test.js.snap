// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`octicon Octicon: name .html(options) return an SVG with invalid options 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG which scales 1`] = `"<svg version=\\"1.1\\" width=\\"32\\" height=\\"32\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG which scales with 0 1`] = `"<svg version=\\"1.1\\" width=\\"0\\" height=\\"0\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG which scales with a fractional value 1`] = `"<svg version=\\"1.1\\" width=\\"19.75\\" height=\\"19.75\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG which scales with invalid parameters 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG with a label 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" aria-label=\\"a label\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG with an additional class 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name an-additional-class\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG with an invalid class 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .html(options) returns an SVG without any options 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-name\\"><path /></svg>"`;

exports[`octicon Octicon: name .keywords() returns keywords 1`] = `undefined`;

exports[`octicon Octicon: name .path(camel) returns camel-case path 1`] = `Object {}`;

exports[`octicon Octicon: name .path(camel) returns path 1`] = `Object {}`;
